packages/client/hmi-client/src/components/widgets/tera-csv-table.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+	<p v-if="isLoading">Loading CSV...</p>
+	<p v-else-if="error">Error loading CSV: {{ error }}</p>
+	<DataTable
+		v-else
+		:size="'small'"
+		showGridlines
+		stripedRows
+		removableSort
+		:value="csvData"
+		:globalFilter="true"
+		:sortMode="'multiple'"
+	>
+		<Column v-for="col in columns" :key="col.field" :field="col.field" :header="col.header" :sortable="true" />
+	</DataTable>
+</template>
+
+<script setup lang="ts">
+import { onMounted, ref } from 'vue';
+import { isEmpty } from 'lodash';
+import DataTable from 'primevue/datatable';
+import Column from 'primevue/column';
+import Papa from 'papaparse';
+
+// Define props for the component
+const props = defineProps({
+	csvText: {
+		type: String,
+		required: true
+	}
+});
+
+const csvData = ref([]);
+const columns = ref([]);
+const isLoading = ref(true);
+const error = ref(null);
+
+function rowReplaceEmptyKey(row: any) {
+	Object.keys(row).forEach((key) => {
+		if (isEmpty(key)) {
+			row.empty = row[key];
+			delete row[key];
+		}
+	});
+	return row;
+}
+
+function onComplete(results: any) {
+	// If the CSV has an empty header, just use 'empty' as the field name
+	columns.value = results.meta.fields.map((field: any) => ({ field: isEmpty(field) ? 'empty' : field, header: field }));
+	csvData.value = results.data.map(rowReplaceEmptyKey);
+	isLoading.value = false;
+}
+
+function onError(err: any) {
+	error.value = err.message;
+	isLoading.value = false;
+}
+
+// Configuration for PapaParse
+// https://www.papaparse.com/docs#config
+const config = {
+	header: true,
+	dynamicTyping: true,
+	skipEmptyLines: true,
+	complete: onComplete,
+	error: onError
+};
+
+// Load CSV when component is mounted
+onMounted(() => {
+	try {
+		Papa.parse(props.csvText, config);
```

Comment:
We already have a handful of csv parsers in house
- D3 has capabilities to parse csv and tsf
- We built an in-house `util/dsv.ts#dsvParse` to be a bit more intelligent for parsing matrices in csv/tsf format and numerical values

Since we are a bit short on time - I think we should flag this as to be consolidated. We don't need another CSV parsing library.

Reply:
Nice, sorry, `dsv` was not what I was looking for in the codebase. I will take a look and see if I can refactor.

Reply:
I will not solve this, as DSV is hard coded for numbers only, and the object returned is not really useful to be displayed to the user.
I could add more feature to DSV to handle that use case, but this is out of scope of this PR...